src/Microsoft.Health.Dicom.Core/Features/Store/StoreValidatorResultBuilder.cs

```diff
@@ -0,0 +1,89 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.ObjectModel;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.Store;
+
+internal sealed class StoreValidatorResultBuilder
+{
+    public StoreValidatorResultBuilder()
+    {
+        ErrorMessages = new Collection<string>();
+        WarningMessages = new Collection<string>();
+        WarningCodes = ValidationWarnings.None;
+        FirstException = null;
+    }
+
+    private Collection<string> ErrorMessages { get; }
+
+    private Collection<string> WarningMessages { get; }
+
+    private ValidationWarnings WarningCodes { get; set; }
```

Comment:
What is the point of these codes? It is a bit confusing because we don't really record the exceptions, just messages, but we preserve these codes in addition to their corresponding messages?

Reply:
We are keeping the "WarningCodes" to continue to support the current behavior.

Reply:
Looking at the current behavior, they appear to be used only to populate more warnings. Now that we have a builder, can't we have these warnings added directly without having the codes as a middleman?